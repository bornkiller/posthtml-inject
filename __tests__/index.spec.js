/**
 * @description - main test suits
 * @author - bornkiller <hjj491229492@hotmail.com>
 */

// Native
const fs = require('fs');
const path = require('path');

// External
const posthtml = require('posthtml');

// Internal
const inject = require('../lib');

// Scope
const readOptions = { encoding: 'utf8' };

describe('posthtml inject plugin', () => {
  it('should validate pass options', () => {
    expect(() => inject()).toThrow();
    expect(() => inject({ elements: [] })).toThrow();
  });

  it('standard case', () => {
    const html = fs.readFileSync(path.resolve(__dirname, '__fixture__', 'standard.html'), readOptions);
    const exclude = 'critical';
    const elements = {
      manifest: [
        {
          tag: 'link',
          attrs: {
            rel: 'manifest',
            href: 'manifest.json'
          }
        }
      ],
      library: [
        {
          tag: 'script',
          attrs: {
            src: 'https://cdn.bootcss.com/react/15.6.1/react.js'
          }
        },
        {
          tag: 'script',
          attrs: {
            src: 'https://cdn.bootcss.com/moment.js/2.18.1/moment.min.js'
          }
        }
      ]
    };

    return posthtml([inject({ exclude, elements })])
      .process(html)
      .then((res) => {
        expect(res.html).toMatchSnapshot();
      });
  });
});
