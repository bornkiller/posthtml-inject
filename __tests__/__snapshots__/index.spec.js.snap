// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posthtml inject plugin standard case 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"UTF-8\\">
  <title>Posthtml</title>
  <!-- slot manifest -->
  <link rel=\\"manifest\\" href=\\"manifest.json\\">
  <!-- slot manifest -->

  <!-- slot empty -->
  <template data-id=\\"posthtml-inject\\" data-name=\\"critical\\"></template>
  <!-- slot empty -->
</head>
<body>
  <!-- slot library CDN scripts -->
  <script src=\\"https://cdn.bootcss.com/react/15.6.1/react.js\\"></script>
  <script src=\\"https://cdn.bootcss.com/moment.js/2.18.1/moment.min.js\\"></script>
  <!-- slot library CDN scripts -->
</body>
</html>
"
`;
